src/bpfprogram.cpp

```diff
@@ -81,17 +82,31 @@ void BpfProgram::set_attach_target(const Probe &probe,
     btf_fun = fun;
   }
 
-  if (btf.get_btf_id(btf_fun, mod, btf_kind) < 0) {
-    const std::string msg = "No BTF found for " + attach_target + ".";
+  std::string err_msg;
+
+  if ((probe.type == ProbeType::fentry || probe.type == ProbeType::fexit) &&
+      mod == "bpf") {
+    attach_target = fun;
+    attach_fd_ = std::move(util::get_fd_for_bpf_prog(fun));
+    if (attach_fd_ < 0) {
+      err_msg = "No BPF program found with name: " + fun + ".";
```

Comment:
I still think you should just make this function return an error, instead of potentially logging something and just returning `void`. It just feels like an anti-pattern to me, but I won't block for this one thing (most of the rest of my complaints are much improved).